beads_web/src/main/java/com/beads/web/dao/OrderDaoImpl.java

```diff
@@ -15,12 +17,49 @@
   public static final String BEAN_NAME = "web.OrderDaoImpl";
 
   @Override
-  public List<Order> getAllOrder() {
+  public List<Order> getOrdersBySearchCriteria(SearchCriteria searchCriteria) {
     CriteriaBuilder criteriaBuilder = getSession().getCriteriaBuilder();
     CriteriaQuery<Order> criteriaQuery = criteriaBuilder.createQuery(Order.class);
     Root<Order> root = criteriaQuery.from(Order.class);
     root.fetch(Order.ORDER_ITEMS);
+    List<Predicate> predicates1 = buildSearchPredicate(criteriaBuilder, root, searchCriteria);
+    Predicate[] predicates = predicates1.toArray(new Predicate[predicates1.size()]);
     criteriaQuery.select(root);
+    if (predicates.length != 0) {
+      criteriaQuery.where(criteriaBuilder.and(predicates));
+    }
     return getSession().createQuery(criteriaQuery).setMaxResults(MAX_ROW_RESULT).list();
   }
+
+  private List<Predicate> buildSearchPredicate(
+      CriteriaBuilder criteriaBuilder,
+      Root<Order> root,
+      SearchCriteria searchCriteria) {
+
+    List<Predicate> predicates = new ArrayList<>();
+    predicates.clear();
+    if (searchCriteria.isOrderIdNotNull()) {
+      predicates.add(criteriaBuilder.equal(root.get(Order.ID), searchCriteria.getOrderId()));
+    }
+    if (searchCriteria.isEmailNotNull()) {
+      predicates.add(criteriaBuilder.like(root.get(Order.EMAIL),
```

Comment:
hm. seem like matching will happen only from the beginning of the line.
May be we need add `"%"` at the beginning too ?